captum/attr/_utils/batching.py

```diff
@@ -12,6 +12,76 @@
     TensorOrTupleOfTensorsGeneric,
     TupleOrTensorOrBoolGeneric,
 )
+from .approximation_methods import approximation_parameters
+
+
+def _batched_attribution(
+    attr_method,
+    num_examples,
+    internal_batch_size,
+    n_steps,
+    include_endpoint=False,
+    **kwargs
+):
+    """
+    This method applies internal batching to given attribution method, dividing
+    the total steps into batches and running each independently and sequentially,
+    adding each result to compute the total attribution.
+
+    Step sizes and alphas are spliced for each batch and passed explicitly for each
+    call to _attribute.
+
+    kwargs include all argument necessary to pass to each attribute call, except
+    for n_steps, which is computed based on the number of steps for the batch.
+
+    include_endpoint ensures that one step overlaps between each batch, which
+    is necessary for some methods, particularly LayerConductance.
+    """
+    # Number of steps for each batch
+    step_count = internal_batch_size // num_examples
+    assert (
+        step_count > 0
+    ), "Internal batch size must be at least equal to the number of input examples."
+    if include_endpoint:
+        assert (
+            step_count > 1
+        ), "Internal batch size must be at least twice the number of input examples."
```

Comment:
Perhaps explaining `why` here would be better. Also, perhaps defaulting like the previous suggestion would be better

Reply:
Does this mean that if the `n_steps=1` then this will always fail because `internal_batch_size` can only be 1 in that case ?
Perhaps making a recommendation message would be helpful.

Reply:
Yes, that will fail (and probably should), since it would've also failed in the implementation (LayerConductance) where this argument is used, because we take the difference between consecutive steps to estimate gradient, and wouldn't be able to do this with 1 step.